### DISCUSSION TITLE
Context in viewModel

### DISCUSSION THREAD

**Discussion by Amir-yazdanmanesh** [2dust/v2rayNG#2942] [Ideas]
I've noticed that  MainViewModel is tightly coupled with the Android Context, which goes against the best practice of keeping ViewModels independent of any Android framework components. I understand the importance of making ViewModels agnostic to the UI layer for better testability and reusability. I'm open to restructuring my codebase to remove any dependencies on the Context and separating out any view-related logic into appropriate view components.

**Comment by mosayeb-a** [comment]
I completely agree with you because it is hard to read and test. It is not only the viewModel that is very attached to Android components, but other components are also very dependent on each other, and changing one of them affects the other. I think it is a very good suggestion.

**Reply by Amir-yazdanmanesh** [reply]
Yes, you're right. I'll start by decoupling the context from the view model and then submit a merge request. I hope @2dust accepts it :)
